### PR TITLE
sql/mysql: use default attributes when from realm is nil

### PR DESCRIPTION
**Pull request by yonidavidson**

### CODE REVIEW

File: sql/mysql/diff.go [hunk 1]
```diff
@@ -21,12 +21,16 @@ type diff struct{ conn }
...
 	// Charset change.
-	if change := d.charsetChange(from.Attrs, from.Realm.Attrs, to.Attrs); change != noChange {
+	if change := d.charsetChange(from.Attrs, ra, to.Attrs); change != noChange {
 		changes = append(changes, change)
 	}
 	// Collation change.
-	if change := d.collationChange(from.Attrs, from.Realm.Attrs, to.Attrs); change != noChange {
+	if change := d.collationChange(from.Attrs, ra, to.Attrs); change != noChange {
 		changes = append(changes, change)
 	}
```
Comment (a8m) [inline sql/mysql/diff.go:35]: I guess the code is panic here (@zeevmoney also showed me that), but why do you have a schema without a Realm?

File: sql/mysql/diff.go [hunk 2]
```diff
@@ -21,12 +21,16 @@ type diff struct{ conn }
 // SchemaAttrDiff returns a changeset for migrating schema attributes from one state to the other.
 func (d *diff) SchemaAttrDiff(from, to *schema.Schema) []schema.Change {
 	var changes []schema.Change
+	var ra []schema.Attr
+	if from.Realm != nil {
+		ra = from.Realm.Attrs
+	}
```
Comment (a8m) [inline sql/mysql/diff.go:27]:
```suggestion
	var (
		topAttrs []schema.Attr
		changes []schema.Change
	)
	if from.Realm != nil {
		topAttrs = from.Realm.Attrs
	}
```

File: sql/mysql/diff.go (hunk 1 above)
Comment (yonidavidson) [reply sql/mysql/diff.go:35]: because sometimes we diff with schemas that are stored in text mode (and the realm is not marshaled )

**Review by a8m — APPROVED** [review]
💯

**Pull request merged**